ee/packages/federation-matrix/src/events/member.ts

```diff
@@ -85,18 +52,149 @@ async function membershipJoinAction(data: HomeserverEventSignatures['homeserver.
+		return;
+	}
+
+	const roomType = content.membership === 'invite' && content?.is_direct ? 'd' : 'c';
+
+	const createState = stripped_state?.find((state) => state.type === 'm.room.create');
+	const roomOriginDomain = createState?.sender?.split(':')?.pop();
+
+	const roomNameState = stripped_state?.find((state) => state.type === 'm.room.name');
+	const matrixRoomName = roomNameState?.content?.name;
+
+	// if is a DM, use the sender username as the room name
+	// otherwise, use the matrix room name and the room origin domain
+	let roomName: string;
+	if (content?.is_direct) {
+		roomName = sender;
+	} else if (matrixRoomName && roomOriginDomain) {
+		roomName = `${matrixRoomName}:${roomOriginDomain}`;
+	} else {
+		roomName = `${room_id}:${roomOriginDomain}`;
+	}
+
+	// TODO: Consider refactoring to create federated rooms using the Matrix roomId as the Rocket.Chat room name and set the display (visual) name as the fName property.
+	const roomFName = roomName;
+
+	const room = await getOrCreateFederatedRoom(
+		room_id as RoomID,
+		roomFName,
+		roomType,
+		inviterUser._id as UserID,
+		inviterUser.username as UserID,
+	);
+	if (!room) {
+		logger.error(`Room not found or could not be created: ${room_id}`);
+		return;
+	}
+
+	await Room.addUserToRoom(room._id, inviteeUser, inviterUser, {
+		invited: true,
+		federation: { inviteEventId: event_id, inviterUsername: inviterUser.username },
+	});
+
+	logger.info(`Successfully processed invite event ${event_id}`);
+}
+
+async function handleJoin(event: MembershipEvent): Promise<void> {
+	// eslint-disable-next-line @typescript-eslint/naming-convention
+	const { room_id, state_key, event_id } = event;
+
+	const joiningUser = await getOrCreateFederatedUser(state_key as UserID);
+	if (!joiningUser) {
+		logger.error(`Failed to get or create joining user: ${state_key}`);
+		return;
+	}
+
+	const room = await Rooms.findOne({ 'federation.mrid': room_id });
+	if (!room) {
+		logger.warn(`Join event for unknown room: ${room_id} - user may be joining before room creation event received`);
+		return membershipJoinAction(event);
+	}
+
+	const subscription = await Subscriptions.findOne({
+		'rid': room._id,
+		'u._id': joiningUser._id,
+	});
+
+	if (!subscription) {
+		logger.info(`User ${state_key} joining room ${room_id} directly (no prior invite)`);
+		return membershipJoinAction(event);
+	}
+
+	logger.info(`User ${state_key} accepting invite to room ${room_id}`);
+
+	await Subscriptions.updateOne(
+		{ _id: subscription._id },
+		{
+			$unset: {
+				'invited': 1,
+				'federation.inviteEventId': 1,
+				'federation.inviterUsername': 1,
+			},
+			$set: {
+				open: true,
+				alert: false,
+				_updatedAt: new Date(),
+			},
+		},
+	);
+
+	logger.info(`Successfully processed join event ${event_id}`);
+}
+
+async function handleLeave(event: MembershipEvent): Promise<void> {
+	// eslint-disable-next-line @typescript-eslint/naming-convention
```

Comment:
you have 2 options here, destructure following the conventions, or you can use the property without destructure ;). but not disable lint rules